test/fiber/test_mutex.rb

```diff
@@ -217,4 +217,24 @@ def test_mutex_deadlock
     thread.join
     RUBY
   end
+
+  def test_mutex_fiber_deadlock_no_scheduler
+    thr = Thread.new do
+      loop do
+        sleep 1
+      end
+    end
+
+    mutex = Mutex.new
+    mutex.synchronize do
+      error = assert_raise ThreadError do
+        Fiber.new do
+          mutex.lock
+        end.resume
+      end
+      assert_includes error.message, "deadlock; lock already owned by another fiber belonging to the same thread"
+    end
+  ensure
+    thr&.kill
```

Comment:
Should also join the thread otherwise it can still be alive in the following tests. `kill` doesn't wait for it.